cmd/zc_traverser_s3.go

```diff
@@ -64,13 +64,25 @@ func (t *s3Traverser) isDirectory(isSource bool) bool {
 }
 
 func (t *s3Traverser) traverse(preprocessor objectMorpher, processor objectProcessor, filters []objectFilter) (err error) {
+	invalidAzureBlobName := func (objectKey string) bool {
+		/* S3 object name is invalid if it ends with period or
+		   one of virtual directories in path ends with period.
+		   This list is not exhaustive
+		 */
+		return strings.HasSuffix(objectKey, ".") ||
+		       strings.Contains(objectKey, "/.")
```

Comment:
Would it make sense to handle these invalid names by removing the invalid chars? Just a thought!

Reply:
Azure cannot compose a PutBlockFromURL if the name matches these patterns. Changing the name will still result in invalid URL.

Reply:
Makes sense!

Reply:
I feel like we _should_ eventually prompt for a rename instead of straight rejecting it. It'd be a nicer UX imo. @zezha-msft thoughts?

Reply:
It's a great question @adreed-msft, unfortunately as @nakulkar-msft found out, our service API is unable to target the source object with the invalid chars. It is a service limitation.

Reply:
Duh-doy! I knew this was related, but forgot that it was the source not the destination.